### PR TITLE
fix(clients): safer replaceAllObjects + metis compliant

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-2407

### Changes included:

Add metis compliant implementation for `replaceAllObjects`, also ensure it never hangs if the source index does not exist, detailed explanation in the docs.

suggested implementation from @Jerska

### CODE REVIEW

**Review by millotp — COMMENTED** [review]
looks really good !

File: website/docs/contributing/add-new-api-client.md [hunk 1]
```diff
@@ -178,14 +178,54 @@ For more information, see [Common Test Suite](../contributing/testing/common-tes
...
+
+:::info
+
+This section explains the decision over the implementation of the helper, as it's quite tricky the get the order right
+
+:::
+
+:::caution
+
+Always wait on the destination index, because:
```
[the remark this answers is not in the document]
Comment (shortcuts) [reply website/docs/contributing/add-new-api-client.md:204]: great point

File: website/docs/contributing/add-new-api-client.md [hunk 2]
```diff
@@ -178,14 +178,54 @@ For more information, see [Common Test Suite](../contributing/testing/common-tes
...
+| `generateSecuredApiKey` | Given a `indexName` and an array of `restrictions`, generates a secured API Key using the SHA-256 algorithm                                                                               | `none`                             | `none`                                                  | [PHP](https://github.com/algolia/api-clients-automation/blob/main/clients/algoliasearch-client-php/lib/Api/SearchClient.php#L2869)                                |
+| `chunkedBatch`          | Creates chunks of `objects` in order to make them fit in multiple `batch` requests                                                                                                        | `none`                             | `none`                                                  | [PHP](https://github.com/algolia/api-clients-automation/blob/main/clients/algoliasearch-client-php/lib/Api/SearchClient.php#L2869)                                |
```
[the remark this answers is not in the document]
Comment (shortcuts) [reply website/docs/contributing/add-new-api-client.md:192]: ah I forgot to update the link

File: website/docs/contributing/add-new-api-client.md [hunk 3]
```diff
@@ -178,14 +178,54 @@ For more information, see [Common Test Suite](../contributing/testing/common-tes
...
+
+This section explains the decision over the implementation of the helper, as it's quite tricky the get the order right
+
+:::
+
+:::caution
+
+Always wait on the destination index, because:
+- it's the only index that we know for sure will exist (since we create it).
+- it is Metis compliant, no need for custom implementation.
```
[the remark this answers is not in the document]
Comment (shortcuts) [reply website/docs/contributing/add-new-api-client.md:206]: _I think_ it's fine, pretty niche documentation, I can ask

**Review by millotp — APPROVED** [review]

**Pull request merged**